gravitee-gateway-http-client/gravitee-gateway-http-client-core/src/main/java/io/gravitee/gateway/http/core/endpoint/EndpointLifecycleManagerFactory.java

```diff
@@ -28,12 +28,12 @@
 public class EndpointLifecycleManagerFactory extends AbstractAutowiringFactoryBean<EndpointLifecycleManager> {
 
     @Autowired
-    private Api api;
+    private GatewayConfiguration gatewayConfiguration;
 
     @Override
     protected EndpointLifecycleManager doCreateInstance() throws Exception {
-        if (api.getProxy().isMultiTenant()) {
-            return new MultiTenantAwareEndpointLifecycleManager();
+        if (gatewayConfiguration.tenant().isPresent()) {
```

Comment:
On a gteway without tenant, the consequence is that we load all endpoints, even if they have tenants.